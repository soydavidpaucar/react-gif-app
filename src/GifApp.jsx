export const GifApp = () => {
  return (
    <>
      <h1>GifApp</h1>
    </>
  );
};
