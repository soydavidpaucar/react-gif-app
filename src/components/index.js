export * from './AddCategory.jsx';
export * from './GifGrid.jsx';
export * from './GifItem.jsx';