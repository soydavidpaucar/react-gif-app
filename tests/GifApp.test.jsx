import { describe, expect, it } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';

import { GifApp } from '../src/GifApp';

describe('test GifApp component', () => {
  it('should match GifApp component with Snapshot', () => {
    const {container} = render(<GifApp />);
    
    expect(container).toMatchSnapshot();
  });
  
  it('should add a new category', () => {
    const {container} = render(<GifApp />);
    
    const input = screen.getByRole('textbox');
    const form = screen.getByLabelText('form');
    
    fireEvent.change(input, {target: {value: 'Roger Federer'}});
    fireEvent.submit(form);
    
    expect(input.value).toBe('');
    expect(container).toMatchSnapshot();
  });
  
  it('should not add a new category if the input is empty', () => {
    const {container} = render(<GifApp />);
    
    const form = screen.getByLabelText('form');
    
    fireEvent.submit(form);
    
    expect(container).toMatchSnapshot();
  })
  
  it('should not add a new category if the category already exists', () => {
    const {container} = render(<GifApp />);
    
    const input = screen.getByRole('textbox');
    const form = screen.getByLabelText('form');
    
    fireEvent.change(input, {target: {value: 'Roger Federer'}});
    fireEvent.submit(form);
    
    fireEvent.change(input, {target: {value: 'Roger Federer'}});
    fireEvent.submit(form);
    
    expect(input.value).toBe('');
    expect(container).toMatchSnapshot();
  })
  
});