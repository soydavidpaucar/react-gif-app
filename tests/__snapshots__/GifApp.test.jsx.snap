// Vitest Snapshot v1

exports[`test GifApp component > should add a new category 1`] = `
<div>
  <main
    class="bg-[#0a192f] p-[60px] min-h-screen"
  >
    <h1
      class="text-[#ccd6f6] text-6xl mb-5 font-black"
    >
      GifApp
    </h1>
    <form
      aria-label="form"
    >
      <input
        class="border border-[#64ffda] rounded px-3 py-2 text-[#ccd6f6] outline-none bg-transparent mb-16 w-full placeholder-[#ccd6f6]"
        placeholder="Search gifs..."
        type="text"
        value=""
      />
    </form>
    <h3
      class="text-[#ccd6f6] text-4xl mb-2 font-semibold"
    >
      Roger Federer
    </h3>
    <p
      class="text-[#ccd6f6] text-1xl"
    >
      Loading...
    </p>
    <div
      class="text-[#8791af] mb-12 flex flex-row flex-wrap justify-center"
    />
  </main>
</div>
`;

exports[`test GifApp component > should match GifApp component with Snapshot 1`] = `
<div>
  <main
    class="bg-[#0a192f] p-[60px] min-h-screen"
  >
    <h1
      class="text-[#ccd6f6] text-6xl mb-5 font-black"
    >
      GifApp
    </h1>
    <form
      aria-label="form"
    >
      <input
        class="border border-[#64ffda] rounded px-3 py-2 text-[#ccd6f6] outline-none bg-transparent mb-16 w-full placeholder-[#ccd6f6]"
        placeholder="Search gifs..."
        type="text"
        value=""
      />
    </form>
  </main>
</div>
`;

exports[`test GifApp component > should not add a new category if the category already exists 1`] = `
<div>
  <main
    class="bg-[#0a192f] p-[60px] min-h-screen"
  >
    <h1
      class="text-[#ccd6f6] text-6xl mb-5 font-black"
    >
      GifApp
    </h1>
    <form
      aria-label="form"
    >
      <input
        class="border border-[#64ffda] rounded px-3 py-2 text-[#ccd6f6] outline-none bg-transparent mb-16 w-full placeholder-[#ccd6f6]"
        placeholder="Search gifs..."
        type="text"
        value=""
      />
    </form>
    <h3
      class="text-[#ccd6f6] text-4xl mb-2 font-semibold"
    >
      Roger Federer
    </h3>
    <p
      class="text-[#ccd6f6] text-1xl"
    >
      Loading...
    </p>
    <div
      class="text-[#8791af] mb-12 flex flex-row flex-wrap justify-center"
    />
  </main>
</div>
`;

exports[`test GifApp component > should not add a new category if the input is empty 1`] = `
<div>
  <main
    class="bg-[#0a192f] p-[60px] min-h-screen"
  >
    <h1
      class="text-[#ccd6f6] text-6xl mb-5 font-black"
    >
      GifApp
    </h1>
    <form
      aria-label="form"
    >
      <input
        class="border border-[#64ffda] rounded px-3 py-2 text-[#ccd6f6] outline-none bg-transparent mb-16 w-full placeholder-[#ccd6f6]"
        placeholder="Search gifs..."
        type="text"
        value=""
      />
    </form>
  </main>
</div>
`;
